src/map.rs

```diff
@@ -266,29 +261,37 @@ impl<K: Key, V: Val<A>, A: Actor> Map<K, V, A> {
     }
 
     /// Update a value under some key, if the key is not present in the map,
-    /// the updater will be given the result of V::default().
-    pub fn update<F, I>(&self, key: I, ctx: AddCtx<A>, f: F) -> Op<K, V, A>
+    /// the updater will use the given `default` value.
+    pub fn update_with_default<F>(&self, key: K, ctx: AddCtx<A>, default: V, f: F) -> Op<K, V, A>
```

Comment:
I'm not convinced this is safe, if `V` is not the V::default, then replicas relying on op based replication will not have the op's building up to this CRDT that was passed in as the default.

The default choice for a missing key needs to be coordinated so that both the `apply` method and the `update` method understand the default value to use. Otherwise this will break convergence.

Reply:
Right, this was an exercise in me getting to know the library. I see the point now.

When I first came along I was confused as to why there was no `insert` method on `Map`. I think something I missed originally was that the `Val` needs to be a CRDT itself as well.

This is all clear now, so I'll revert this change and maybe add some prose as to why this is the case.

Thanks for the explanation :grin: